blocks/market-insight/market-insight.js

```diff
@@ -0,0 +1,209 @@
+  discoverMoreDiv.append(aLink);
+  return discoverMoreDiv;
+}
+
+function createSocialButton(button, block) {
+  const link = button.closest('li').querySelector('a').href;
+  const encodeLink = encodeURIComponent(link);
+  const encodeTitle = encodeURIComponent(document.title);
+  const socialMap = new Map([
+    ['whatsapp', `https://api.whatsapp.com/send?text=Hey! Check out this: ${encodeLink}`],
+    ['facebook', `http://www.facebook.com/sharer.php?u=${encodeLink}&t=${encodeTitle},'sharer',toolbar=0,status=0,width=626,height=436`],
+    ['linkedin', `https://www.linkedin.com/sharing/share-offsite/?url=${link}`],
+    ['twitter', `https://twitter.com/intent/tweet?url=${link}`],
+  ]);
+
+  const alinks = block.querySelectorAll('.modal .modal-body li a');
+  if (alinks && alinks.length > 0) {
+    [...alinks].forEach((alink) => {
+      if (!alink.classList.contains('copy-link')) {
+        alink.href = socialMap.get(alink.className);
+      }
+    });
+  }
+}
+
+function addSocialButtonEvent(button, block) {
+  button.addEventListener('click', () => {
+    const modal = block.querySelector('.modal');
+    if (modal) {
+      createSocialButton(button, block);
+      modal.classList.toggle('visible');
+      const body = document.querySelector('body');
+      body.classList.toggle('modal-open');
+    }
+  });
+}
+
+function decorateCards(results, blockCfg, block) {
+  const powerBy = (blockCfg['power-by'] ?? '').trim().toLowerCase();
+  const publishedOn = (blockCfg['published-on'] ?? '').trim().toLowerCase();
+  const ul = createDiv('ul', '');
+  // show 3 cards by default
+  for (let index = 0; index < (results.length > 3 ? 3 : results.length); index += 1) {
+    const result = results[index];
+    const li = createDiv('li', '');
+    const title = createDiv('div', 'cards-title');
+    const description = createDiv('div', 'cards-description');
+    const powerby = createDiv('div', 'cards-powerby');
+    // Cards title
+    const h3 = createDiv('h3', '');
+    const aLink = createDiv('a', '');
+    aLink.href = result.link;
+    aLink.target = '_blank';
+    aLink.append(result.title);
+    h3.append(aLink);
+    title.append(h3);
+    // Cards description
+    description.innerHTML = decodeURIComponent(result.description);
+    // Cards powerby
+    const powerbyDiv = createDiv('div', '');
+    const powerbyContent = createDiv('p', '');
+    powerbyContent.textContent = powerBy;
+    const publishedOnContent = createDiv('p', '');
+    const publishedon = result.publishedon.replaceAll(' ', '-');
+    publishedOnContent.textContent = `${publishedOn} ${publishedon}`;
+    powerbyDiv.append(powerbyContent);
+    powerbyDiv.append(publishedOnContent);
+    powerby.append(powerbyDiv);
+    // Social share button
+    const socialShare = createDiv('div', 'socialshare');
+    const button = createDiv('button', '');
+    const image = createDiv('img', '');
+    image.src = '../../icons/gray-share-icon.svg';
```

Comment:
we can `decorateIcons` from aem.js

Reply:
The `decorateIcons` didn't add image alt properly, let me make a small change.